Providers/Profile.swift

```diff
@@ -503,44 +498,40 @@ public class BrowserProfile: Profile {
         self.remoteClientsAndTabs.insertCommands(commands, forClients: clients) >>> { self.syncManager.syncClients() }
     }
 
-    lazy var logins: protocol<BrowserLogins, SyncableLogins, ResettableSyncStorage> = {
+    lazy var logins: BrowserLogins & SyncableLogins & ResettableSyncStorage = {
         return SQLiteLogins(db: self.loginsDB)
     }()
 
     // This is currently only used within the dispatch_once block in loginsDB, so we don't
     // have to worry about races giving us two keys. But if this were ever to be used
     // elsewhere, it'd be unsafe, so we wrap this in a dispatch_once, too.
-    private var loginsKey: String? {
+    lazy private var loginsKey: String? = {
```

Comment:
I don't think this is equivalent to the `dispatch_once` used in the Swift 3 code. This would imply that the `loginsKey` and `loginsDB` objects would be created lazly _per an instance_. What we want to do here is use `static`.

I started to write a patch to add this but ran into an interesting problem. Within the `dispatch_once` block we reference `self.files`. This means that previously we were binding the first created file accessor to our loginsDB which is definitely not correct. I'm a bit wary of just making this a `static` also because the current assumption is we create one per a profile on purpose.

@rnewman @farhanpatel @fluffyemily Any thoughts?